compiler/rustc_hir_analysis/src/autoderef.rs

```diff
@@ -69,6 +73,14 @@ impl<'a, 'tcx> Iterator for Autoderef<'a, 'tcx> {
         }
 
         // Otherwise, deref if type is derefable:
+        // NOTE: in the case of self.use_receiver_trait = true, you might think it would
+        // be better to skip this clause and use the Overloaded case only, since &T
+        // and &mut T implement Receiver. (They don't implement Deref). We can't do that
```

Comment:
[`&T` does implement `Deref`](https://doc.rust-lang.org/std/ops/trait.Deref.html#impl-Deref-for-%26T). Am I misunderstanding something here?

Reply:
Good spot, this comment is outdated, I'll tweak.

Reply:
I think we should probably just mention that this is an optimization that applies equally to `Receiver` or `Deref`, and the fact that we record those as *builtin* derefs has consequences on const and the emitted MIR.